pepsico/projections/maproom.py

```diff
@@ -168,44 +165,307 @@ def pick_location(n_clicks, click_lat_lng, latitude, longitude):
+            xaxis_title="Time",
+            yaxis_title=f'{data_list[0].attrs["long_name"]} ({units})',
+            title={
+                "text": (
+                    f'{data_list[0]["T"].dt.strftime("%b")[0].values}-'
+                    f'{data_list[0]["seasons_ends"].dt.strftime("%b")[0].values} '
+                    f'{variable} seasonal average from model {model}'
+                ),
+                "font": dict(size=14),
+            }
+        )
+        return local_graph
+
+
+    @APP.callback(
+        Output("map_description", "children"),
+        Input("scenario", "value"),
+        Input("model", "value"),
+        Input("variable", "value"),
+        Input("submit_season", "n_clicks"),
+        Input("submit_projy", "n_clicks"),
+        Input("submit_refy", "n_clicks"),
+        State("start_month", "value"),
+        State("end_month", "value"),
+        State("start_year", "value"),
+        State("end_year", "value"),
+        State("start_year_ref", "value"),
+        State("end_year_ref", "value"),
+    )
+    def write_map_description(
+        scenario,
+        model,
+        variable,
+        n_clicks,
+        py_clicks,
+        ry_clicks,
+        start_month,
+        end_month,
+        start_year,
+        end_year,
+        start_year_ref,
+        end_year_ref,
+    ):
+        return (
+            f'The Map displays the change in {start_month}-{end_month} seasonal average of '
+            f'{variable} from {model} model under {scenario} scenario projected for '
+            f'{start_year}-{end_year} with respect to historical {start_year_ref}-'
+            f'{end_year_ref}'
+        )
+
+
+    @APP.callback(
+        Output("map_title", "children"),
+        Input("scenario", "value"),
+        Input("model", "value"),
+        Input("variable", "value"),
+        Input("submit_season","n_clicks"),
+        Input("submit_projy","n_clicks"),
+        Input("submit_refy","n_clicks"),
+        State("start_month", "value"),
+        State("end_month", "value"),
+        State("start_year", "value"),
+        State("end_year", "value"),
+        State("start_year_ref", "value"),
+        State("end_year_ref", "value"),
+    )
+    def write_map_title(
+        scenario,
+        model,
+        variable,
+        n_clicks,
+        py_clicks,
+        ry_clicks,
+        start_month,
+        end_month,
+        start_year,
+        end_year,
+        start_year_ref,
+        end_year_ref,
+    ):
+        return (
+            f'{start_month}-{end_month} {start_year}-{end_year} '
+            f'{scenario} {model} {variable} change from '
+            f'{start_year_ref}-{end_year_ref}'
+        )
+
+
+    def seasonal_change(
+        scenario,
+        model,
+        variable,
+        region,
+        start_month,
+        end_month,
+        start_year,
+        end_year,
+        start_year_ref,
+        end_year_ref,
+    ):
+        ref = ac.unit_conversion(ac.seasonal_data(
```

Comment:
I suggest doing the unit conversion just once, where we read in the data, rather than doing it in every place where we use the data.